### PR TITLE
Create VecWorker

### PR DESCRIPTION
**Pull request by krzentner**
This allows vectorizing policy evaluation in new-style samplers (as long
as no other custom worker is needed).

### CODE REVIEW

**Comment by krzentner** [comment]
This change needs a little more testing before I consider it done, but the smoketest passes and it's conceptually complete.

File: src/garage/sampler/vec_worker.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+                                 '({}) environments, but received {} '
+                                 'environments.'.format(
+                                     self._n_envs, len(env_update)))
+            for env_index, env_up in enumerate(env_update):
+                self._udpate_env_inner(env_up, env_index)
+        elif env_update is not None:
+            for env_index in range(self._n_envs):
+                self._udpate_env_inner(copy.deepcopy(env_update), env_index)
+
+    def _udpate_env_inner(self, env_update, env_index):
```
Comment (ryanjulian) [inline src/garage/sampler/vec_worker.py:91]: update?

File: src/garage/sampler/vec_worker.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        if isinstance(env_update, EnvUpdate):
+            self._envs[env_index] = env_update(self._envs[env_index])
+            self._needs_env_reset = True
+        elif isinstance(env_update, gym.Env):
+            if self._envs[env_index] is not None:
+                self._envs[env_index].close()
+            self._envs[env_index] = env_update
+            self._needs_env_reset = True
+        else:
+            raise TypeError('Uknown environment update type.')
```
Comment (ryanjulian) [inline src/garage/sampler/vec_worker.py:113]: Unknown

File: src/garage/sampler/vec_worker.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    Alternates between taking a single step in all environments and asking the
+    policy for an action for every environment. This allows computing a batch
+    of actions, which is generally much more efficient than computing a single
+    action when using neural networks.
+
+    Args:
+        seed(int): The seed to use to intialize random number generators.
+        max_path_length(int or float): The maximum length paths which will
+            be sampled. Can be (floating point) infinity.
+        worker_number(int): The number of the worker this update is
```
Comment (naeioi) [inline src/garage/sampler/vec_worker.py:25]: I feel like this is better named `worker_index`

File: src/garage/sampler/vec_worker.py (hunk 1 above)
Comment (krzentner) [reply src/garage/sampler/vec_worker.py]: Thanks!

File: src/garage/sampler/vec_worker.py (hunk 3 above)
Comment (krzentner) [reply src/garage/sampler/vec_worker.py:25]: Yeah, maybe. But it's out of scope for this PR either way, since that's part of the worker API.

**Comment by ryanjulian** [comment]
@Mergifyio rebase

**Pull request merged**